app/src/main/java/com/github/onlynotesswent/model/users/UserRepositoryFirestore.kt

```diff
@@ -71,11 +71,13 @@ class UserRepositoryFirestore(private val db: FirebaseFirestore) : UserRepositor
   }
 
   override fun updateUser(user: User, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection(collectionPath)
-        .document(user.uid)
-        .set(user)
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { exception -> onFailure(exception) }
+    deleteUserById(
+        user.uid,
+        onSuccess = {
+          addUser(user, onSuccess, onFailure)
+          onSuccess()
```

Comment:
Remove OnSuccess call, this will lead to failure if the delete user is successful but add user fails (the user will be deleted instead of being updated